pkg/sdkserver/sdkserver.go

```diff
@@ -824,3 +833,25 @@ func (s *SDKServer) updateConnectedPlayers(ctx context.Context) error {
 	s.recorder.Event(gs, corev1.EventTypeNormal, "PlayerCount", fmt.Sprintf("Set to %d", gs.Status.Players.Count))
 	return nil
 }
+
+// NewSDKServerContext returns a Context that cancels when SIGTERM or os.Interrupt
+// is received and the GameServer's Status is shutdown
+func (s *SDKServer) NewSDKServerContext(ctx context.Context) context.Context {
+	ctx, cancel := context.WithCancel(ctx)
+	c := make(chan os.Signal, 2)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-c
```

Comment:
Random thought:

Any reason why you didn't take the `ctx` from `NewSigKillContext` and then wrap it in your own ctx and wait for Shudown to happen after it was `Done()` ?

Probably a bit easier to test too.

Reply:
thanks for the suggestion! thats much better :)